subsys/settings/src/settings_store.c

```diff
@@ -155,6 +152,14 @@ int settings_save_one(const char *name, void *value, size_t val_len)
 		return -ENOENT;
 	}
 
+	if (val_len > 0 && value == NULL) {
+		return -EINVAL;
+	}
+
+	if (val_len == 0) {
```

Comment:
It is not required as all checks are done using lengths compassion.

Reply:
> Correction is not required as all checks are done using lengths comparasion.

@nvlsianpu good point. PR has been updated - now the first patch only adds the -EINVAL return.